DFC.ServiceTaxonomy.GraphSync/Controllers/HomeController.cs

```diff
@@ -1,9 +1,50 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System;
+using System.IO;
+using System.Text;
+using DFC.ServiceTaxonomy.GraphSync.Services.Interface;
+using Microsoft.AspNetCore.Mvc;
 
 namespace DFC.ServiceTaxonomy.GraphSync.Controllers
 {
     public class HomeController : Controller
     {
+        private readonly ISynonymService _synonymService;
+
+        public HomeController(ISynonymService synonymService)
+        {
+            _synonymService = synonymService ?? throw new ArgumentNullException(nameof(synonymService));
```

Comment:
The null check is not really necessary, as the controller won't be created unless `ISynonymService` is available (unless the controller is being used outside of asp.net I suppose). Not that the check will do any harm.

Reply:
As this is DId, if there is no service collection entry this should throw? Or is it because it's the only parameter in the constructor?

Reply:
If there's no `ISynonymService` registered, the controller will fail to be injected before the constructor is even called, so `synonymService` should never be null.